specification/documentation/callbacks/message-subscription-openapi-spec.json

```diff
@@ -0,0 +1,230 @@
+        "operationId": "post-v1-callbacks",
+        "parameters": [
+          {
+            "name": "x-hmac-sha256-signature",
+            "in": "header",
+            "description": "Contains a HMAC-SHA256 signature of the request body using a pre-agreed secret",
+            "schema": {
+              "type": "string",
+              "examples": [
+                "9ee8c6aab877a97600e5c0cd8419f52d3dcdc45002e35220873d11123db6486f"
+              ]
+            }
+          },
+          {
+            "name": "x-api-key",
+            "in": "header",
+            "description": "Contains the pre-agreed API key.",
+            "schema": {
+              "type": "string",
+              "examples": [
+                "0bb04a0e-d005-42dd-8993-dacf37410a12"
+              ]
+            }
+          }
+        ],
+        "requestBody": {
+          "content": {
+            "application/vnd.api+json": {
+              "schema": {
+                "type": "object",
+                "properties": {
+                  "data": {
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                      "$ref": "#/components/schemas/CallbackRequest"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        },
+        "responses": {
+          "202": {
+            "description": "Accepted"
+          },
+          "401": {
+            "description": "Unauthorized"
+          },
+          "403": {
+            "description": "Forbidden"
+          }
+        },
+        "servers": [
+          {
+            "url": "http://localhost:3000"
+          }
+        ]
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "CallbackRequest": {
+        "type": "object",
+        "title": "CallbackRequest",
+        "properties": {
+          "type": {
+            "const": "MessageStatus"
+          },
+          "attributes": {
+            "type": "object",
+            "properties": {
+              "messageId": {
+                "$ref": "#/components/schemas/Type_KSUID"
+              },
+              "messageReference": {
+                "type": "string",
+                "format": "uuid",
+                "examples": [
+                  "1642109b-69eb-447f-8f97-ab70a74f5db4"
+                ]
+              },
+              "messageStatus": {
+                "$ref": "#/components/schemas/Enum_MessageStatus"
+              },
+              "messageStatusDescription": {
+                "type": "string"
+              },
+              "channels": {
+                "type": "array",
+                "minItems": 1,
+                "items": {
+                  "$ref": "#/components/schemas/MessageStatusChannel"
+                }
+              },
+              "timestamp": {
+                "type": "string",
+                "description": "Timestamp of when the callback was generated.",
```

Comment:
This should be the timestamp of when the dynamodb record was inserted/updated.

Reply:
Its the same thing, the time at which the message was created/inserted is also the time this callback would be generated. I would say its more user friendly to keep it like this to allow consumers to order their callbacks which we don't support.

Reply:
They are roughly equal, but only when things are running smoothly. If I was receiving these messages I would want to know the timestamp from dynamodb not when the callback request was created.
It's a small change which I've squeezed CCM-3548.